### PR TITLE
Add editor menu when double-click on Editor and fix some bugs

### LINKED ISSUE #370: the back button on my android phone doesn't work

### LINKED ISSUE #396: can't paste into editor on mobile
double tapping into the editor or long tapping the editor does not show the option to paste text on mobile

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #396
Resolve #370 

This PR :
1. Add option menu for editor like copy, paste
2. Fix suggestion of editor in restore seed page wrong position
3. Add logic for back button on android device

Screenshot

image

### CODE REVIEW

File: ui/cryptomaterial/editor.go [hunk 1]
```diff
@@ -154,19 +161,47 @@ func (t *Theme) Editor(editor *widget.Editor, hint string) Editor {
...
+		if len(clicks) > 0 {
+			clk := clicks[len(clicks)-1]
+			if clk.NumClicks == 2 {
+				e.isShowMenu = true
+			}
+			if clk.NumClicks != 2 && clk.NumClicks > 0 {
+				e.isShowMenu = false
+			}
+		}
+	}
```
Comment (ukane-philemon) [inline ui/cryptomaterial/editor.go:200]: double `if` check.

File: ui/cryptomaterial/editor.go [hunk 2]
```diff
@@ -355,6 +415,25 @@ func (e Editor) handleEvents() {
...
+
+	if e.paste.Clicked() {
+		clipboard.ReadOp{Tag: &e.eventKey}.Add(gtx.Ops)
+		e.isShowMenu = false
+	}
+}
+
+func (e *Editor) processEvent(gtx C) {
+	for _, ke := range gtx.Events(&e.eventKey) {
```
Comment (ukane-philemon) [inline ui/cryptomaterial/editor.go:431]: `ke` to `eventKey` or `key`?

File: ui/page/components/seed_restore_page.go [hunk 3]
```diff
@@ -274,20 +274,16 @@ func (pg *SeedRestore) seedEditorViewMobile(gtx layout.Context) layout.Dimension
...
-				)
-			})
-		}),
-	)
+	return l.Layout(gtx, len, func(gtx C, i int) D {
+		return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+			layout.Rigid(func(gtx C) D {
+				return layout.Inset{Bottom: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
+					pg.layoutSeedMenu(gtx, i*5+startIndex)
+					return pg.seedEditors.editors[i*5+startIndex].Layout(gtx)
```
Comment (ukane-philemon) [inline ui/page/components/seed_restore_page.go:282]: Uhmm, what's the rationale for this calculation?

File: ui/cryptomaterial/button.go [hunk 4]
```diff
@@ -113,6 +113,18 @@ func (t *Theme) IconButton(icon *widget.Icon) IconButton {
 	}
 }
 
+func (t *Theme) NewIconButton(icon *widget.Icon, click *widget.Clickable) IconButton {
+	return IconButton{
+		IconButtonStyle{
+			Icon:   icon,
+			Button: click,
+			Size:   unit.Dp(24),
+			Inset:  layout.UniformInset(unit.Dp(12)),
```
Comment (ukane-philemon) [inline ui/cryptomaterial/button.go:122]: Can use `values.MarginPaddingxx` for these yh?

**Review by ukane-philemon — CHANGES_REQUESTED** [review]
Preliminary review. Good work @JustinBeBoy

File: ui/page/components/seed_restore_page.go (hunk 3 above)
Comment (JustinBeBoy) [reply ui/page/components/seed_restore_page.go:282]: that is old code, I just remove some code not need

Comment (JustinBeBoy) [reply ui/page/components/seed_restore_page.go:282]: from current logic, each line has 5 editors

**Review by JoeGruffins — COMMENTED** [review]
The back button is still not working on my phone.

The double click copy/paste is working. I have some nits though, feel free to ignore but stating anyhow. I cannot seem to select all. Maybe not normally useful because there's not usually a reason to have spaces, but maybe if you were signing a message? Also, I would expect the copy and paste to be click and hold. Unsure if that is possible. Is double clicking a thing usually?

**Comment by JustinBeBoy** [comment]
> ~The back button is still not working on my phone.~ After uninstalling the app entirely and re-installing the back button is working on this commit.
> 
> The double click copy/paste is working. I have some nits though, feel free to ignore but stating anyhow. I cannot seem to select all. Maybe not normally useful because there's not usually a reason to have spaces, but maybe if you were signing a message? Also, I would expect the copy and paste to be click and hold. Unsure if that is possible. Is double clicking a thing usually?
> 
> Actually the copy and paste is not working in the sign message area.

good catch.

For it to work perfectly, we need to improve further, I will do it in the future, double clicks are quite common, so is click and hold, we will improve it later, don't you agree

> Actually the copy and paste is not working in the sign message area.

I will check it, maybe some processes are overlapping

**Review by dreacot — APPROVED** [review]

**Pull request merged**